Specifications/Applications/for_ApplicationStructureFragment/Segment.cs

```diff
@@ -11,9 +12,26 @@ public Segment(string name)
         
         public IApplicationLocationSegmentName Name => throw new NotImplementedException();
 
+        public IEnumerable<IApplicationLocationSegment> Children => throw new NotImplementedException();
```

Comment:
This is a code smell to me.  If we can have NotImplementedException then that's an indication that our abstraction is wrong.  I think I would want to know why we need the interface.  Can we get away with a base Segment class that has default implementations?  It can be abstract and then implemented with concrete types if need be.